api/src/wfl/module/aou.clj

```diff
@@ -47,17 +45,31 @@
    :disk_size                        100
    :preemptible_tries                3})
 
-(defn map-aou-environment
-  "Map AOU-ENV to environment for inputs preparation."
-  [aou-env]
-  ({:aou-dev "dev" :aou-prod "prod"} aou-env))
+(def ^:private known-cromwells
```

Comment:
I really wouldn't bother with this. If you really wanted to test if you were given a valid cromwell, maybe curl the `/engine/v1/version` endpoint and test it looks something like {"cromwell":"52-606efd2"} (i.e. `(contains? response-body :cromwell)`)

Reply:
Actually this is probably a security issue... Forget I said anything.

Reply:
Actually this is a good point for discussion and I really want to get your insights! 

I was talking to @tbl3rd about this and the only reason I used `known-cromwells` here is to prevent "security holes" that I don't know what they are … In fact, `known-cromwells` really prevents us from passing "arbitrary" cromwell/Terra/executor URLs to WFL in runtime, and I want to find a way to support that.

The way it is implemented in the PR is not very different from the `environments.clj`, each module maintains a `Cromwell URL -> default inputs & options` map, choosing what inputs & options to use based on the URLs (like we did for `env` keywords) or throw if unknown Cromwells are given. I'm thinking if we want to loosen the restrictions, when unknown Cromwells are given, some of the workflows, instead of throwing, WFL may just use a set of default inputs & options, such as the one maps to dev cromwell?

Reply:
Yeah. We don't want people to pass in arbitrary URLs for WFL to bash on.

Reply:
Just to clarify, are you talking about passing arbitrary Cromwell URLs from user request at runtime? In that case yes, please restrict it to a list of acceptable URLs only, as allowing arbitrary ones would open the service to a kind of "open redirect" attack.

Reply:
Yeah, so we're … not … gonna do that.